openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -343,6 +366,42 @@ def end_datetime_text(self, format_string="SHORT_DATE"):
             strftime_localized
         )
 
+    @property
+    def sorting_score(self):
+        """
+        Returns a tuple that can be used to sort the courses according
+        the how "new" they are. The "newness" score is computed using a
+        heuristic that takes into account the announcement and
+        (advertised) start dates of the course if available.
+
+        The lower the number the "newer" the course.
+        """
+        return course_metadata_utils.sorting_score(self.start, self.advertised_start, self.announcement)
+
+    @property
+    def start_type(self):
+        """
+        Returns the type of the course's 'start' field.
+        """
+        if self.advertised_start is not None:
+            return u'string'
```

Comment:
Good catch.  Previous code that I moved, but looks like a bug.  Fixed and updated the unit test.